inception/inception-search-mtas/src/main/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasDocumentIndex.java

```diff
@@ -396,6 +402,221 @@ public long numberOfQueryResults(SearchQueryRequest aRequest)
+                if (results.get("max") > 0.0) {
+                    nonNullStats.put(layer.getUiName() + "." + feature.getUiName(), results);
+                }
+
+                results = getLayerStatistics(aStatisticRequest, searchString, fullDocSet, true);
+                results.remove("stats");
+                results.remove("sourceNumber");
+                results.put("Number of Documents", results.get("n"));
+                results.remove("n");
+                allStats.put("per Sentence: " + layer.getUiName() + "." + feature.getUiName(),
+                        results);
+                if (results.get("max") > 0.0) {
+                    nonNullStats.put(
+                            "per Sentence: " + layer.getUiName() + "." + feature.getUiName(),
+                            results);
+                }
+            }
+        }
+        Map<String, Double> results = getLayerStatistics(aStatisticRequest, "<Token=\"\"/>",
+                fullDocSet, false);
+        results.remove("stats");
+        results.remove("sourceNumber");
+        results.put("Number of Documents", results.get("n"));
+        results.remove("n");
+        allStats.put("Token Count", results);
+        nonNullStats.put("Token Count", results);
+
+        results = getLayerStatistics(aStatisticRequest, "<Token=\"\"/>", fullDocSet, true);
+        results.remove("stats");
+        results.remove("sourceNumber");
+        results.put("Number of Documents", results.get("n"));
+        results.remove("n");
+        allStats.put("per Sentence: Token Count", results);
+        nonNullStats.put("per Sentence: Token Count", results);
+
+        results = getLayerStatistics(aStatisticRequest, "<s=\"\"/>", fullDocSet, false);
+        results.remove("stats");
+        results.remove("sourceNumber");
+        results.put("Number of Documents", results.get("n"));
+        results.remove("n");
+        allStats.put("Sentence Count", results);
+        nonNullStats.put("Sentence Count", results);
+
+        return new StatisticsResult(aStatisticRequest, allStats, nonNullStats);
+    }
+
+    @Override
+    public ArrayList<Integer> getUniqueDocuments(StatisticRequest aStatisticRequest)
+        throws IOException
+    {
+        IndexSearcher searcher = null;
+        Map<Long, Long> annotatableDocuments = listAnnotatableDocuments(
+                aStatisticRequest.getProject(), aStatisticRequest.getUser());
+
+        ArrayList<Integer> fullDocSet = new ArrayList<Integer>();
+
+        try {
+            searcher = getSearcherManager().acquire();
+            IndexReader reader = searcher.getIndexReader();
+
+            for (int i = 0; i < reader.maxDoc(); i++) {
+                String sourceID = reader.document(i).get(FIELD_SOURCE_DOCUMENT_ID);
+                String annotationID = reader.document(i).get(FIELD_ANNOTATION_DOCUMENT_ID);
+                if (Long.valueOf(annotationID) != -1L) {
```

Comment:
Can you write 

if (Long.valueOf(annotationID) == -1L) {
continue;
}
?

Reply:
No, I can't. In the case of Long.valueOf(annotationID) == -1L we can not simply jump to the next document. We know then that this is a source document. This will not be included in fullDocSet, unless it has no annotation document linked to it. This special case is checked by the else if statement.